presto-tests/src/main/java/com/facebook/presto/tests/FeatureSelection.java

```diff
@@ -0,0 +1,136 @@
+
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import static java.util.Objects.requireNonNull;
+
+public final class FeatureSelection
+{
+    public static FeatureSelection all()
+    {
+        return allFeatures().build();
+    }
+
+    public static Builder allFeatures()
+    {
+        return new Builder();
+    }
+
+    public static Builder features(TestedFeature... features)
+    {
+        requireNonNull(features);
+        return new Builder().including(features);
+    }
+
+    public static FeatureSelection excluding(TestedFeature... features)
+    {
+        requireNonNull(features);
+        return new Builder().excluding(features);
+    }
+
+    private final Optional<Set<TestedFeature>> includes;
+    private final Optional<Set<TestedFeature>> excludes;
+
+    private FeatureSelection(Optional<Set<TestedFeature>> includes, Optional<Set<TestedFeature>> excludes)
+    {
+        requireNonNull(includes, "includes is null");
+        requireNonNull(excludes, "excludes is null");
+
+        this.includes = includes.map(ImmutableSet::copyOf);
+        this.excludes = excludes.map(ImmutableSet::copyOf);
+    }
+
+    public boolean isSelected(TestedFeature feature)
+    {
+        requireNonNull(feature);
+        return areSelected(ImmutableSet.of(feature));
+    }
+
+    public boolean areSelected(Set<TestedFeature> features)
+    {
+        requireNonNull(features);
+
+        if (excludes.isPresent() && excludes.get().stream().anyMatch(features::contains)) {
+            return false;
+        }
+
+        return !includes.isPresent() || includes.get().stream().anyMatch(features::contains);
+    }
+
+    public static class Builder
+    {
+        private Optional<Set<TestedFeature>> includes = Optional.empty();
+        private Optional<Set<TestedFeature>> excludes = Optional.empty();
+
+        private Builder()
+        {
+        }
+
+        public Builder including(TestedFeature... features)
+        {
+            requireNonNull(features, "features is null");
+            Stream.of(features).forEach(this::include);
+
+            return this;
+        }
+
+        public Builder include(TestedFeature feature)
+        {
+            excludes.ifPresent(excludeGet -> excludeGet.remove(feature));
+
+            if (!includes.isPresent()) {
+                includes = Optional.of(new HashSet<>());
+            }
+
+            includes.get().add(feature);
+
+            feature.getFeatures().forEach(this::include);
+
+            return this;
+        }
+
+        public FeatureSelection excluding(TestedFeature... features)
+        {
+            requireNonNull(features, "features is null");
+            Stream.of(features).forEach(this::exclude);
+
+            return build();
+        }
+
+        public Builder exclude(TestedFeature feature)
+        {
+            includes.ifPresent(includesGet -> includesGet.remove(feature));
```

Comment:
wasn't exception here better? It showed that there is something fishy with configuration (maybe unintended includes/excludes intersection)